buildSrc/plugins/src/main/groovy/org/jetbrains/kotlin/KonanTest.groovy

```diff
@@ -443,9 +443,11 @@ class RunExternalTestGroup extends OldKonanTest {
                     }
                     if (line.contains("$pkg.") && ! (line =~ packagePattern || line =~ importRegex)
                             && ! vars.contains(pkg)) {
-                        def idx = line.indexOf("$pkg")
-                        if (! (idx > 0 && Character.isJavaIdentifierPart(line.charAt(idx - 1))) ) {
-                            line = line.substring(0, idx) + "$sourceName.$pkg" + line.substring(idx + pkg.length())
+                        def idx = 0
+                        while ((idx = line.indexOf("$pkg", idx)) > 0) {
```

Comment:
Isn't this loop infinite if `line` contains `pkg`?

Reply:
Yes, it is. Index increment is missing. Will fix it.

Reply:
Still may happen to be infinite, if `sourceName.length() + 1 >= pkg.length()`.